### ISSUE TITLE
Menu: First item highlighted on opening

### ISSUE THREAD

**Issue by zlovin** [angular/material#3352]
In Chrome (Mac OSX), the first item in an mdMenu is highlighted (focused?) when the menu is opened, and stays highlighted even if other menu items are hovered over with the mouse. The other menu items highlight properly, but the first item does not unhighlight. 

The latest docs on the Angular Material reference show the issue: https://material.angularjs.org/latest/#/demo/material.components.menu

Highlighted on open:
![screen shot 2015-06-18 at 3 06 49 pm](https://cloud.githubusercontent.com/assets/8484077/8240919/bdc218ae-15cb-11e5-923d-b31ce7b83acd.png)

Highlighted when other items are hovered by mouse:
![screen shot 2015-06-18 at 3 07 56 pm](https://cloud.githubusercontent.com/assets/8484077/8240936/d9fe1de2-15cb-11e5-95bc-ffb30cf43572.png)

Using the arrow keys to change the focused item moves the highlight to another item, but if another item is hovered over with the mouse then the newly arrow-key-focused item retains its highlight.

**Comment by crrrazymind** [comment]
+1 Chrome 43.0.2357.125 on Ubuntu

Spec doesn't have a highlight on the icon dropdown.
https://www.google.com/design/spec/components/menus.html#menus-usage

**Comment by marcysutton** [comment]
From #3707: I'm guessing that is because the list of options is in a separate area of the DOM than the triggering element, so focus has to be sent to it for accessibility. Internally in the component, focus could be sent to an element wrapping the options instead (using tabindex="-1").

**Comment by Masadow** [comment]
@marcysutton In #3707, it is related to md-select, so there is no separate area of the DOM in that case. Example generating the focus behaviour : 

            <md-select multiple="true" placeholder="Type d'établissement">
                <md-option>Ecole d'ingénieur</md-option>
                <md-option>Ecole de commerce</md-option>
                <md-option>Ecole spécialisée</md-option>
                <md-option>Université</md-option>
            </md-select>

**Comment by marcoocram** [comment]
+1

**Comment by marcysutton** [comment]
Yes, but look at the compiled source in the DOM. The options are embedded in a completely different location than the `<md-select>` tag. Here is a demo: http://codepen.io/marcysutton/pen/yyxREg?editors=101

Menus and selects work in the same way, so both components need similar focus management.

**Comment by grosem** [comment]
+1

**Comment by AzaroQL** [comment]
+1

**Comment by george3447** [comment]
+1

**Comment by fernandomantoan** [comment]
+1

**Comment by yochayme** [comment]
+1

**Comment by prasannapattam** [comment]
It would be nice, if the current page is highlighted. I am using Menu Bar, I would like to highlight the current page (that belongs to a menu bar) be high lighted when the menu is opened.

**Comment by hynese** [comment]
I'm also having this issue. I understand there are some accessibility concerns that need careful consideration. Forcing the first item to be highlighted doesn't make sense in the application I am building.

I'm using the `angularui:angular-ui-router` package.

@prasannapattam I would recommend seeing if using `ui-sref` and `ui-sref-active` parameters would work for handling active/inactive menu contexts.

**Comment by prasannapattam** [comment]
@hynese, I am building menu from database and I believe I cannot use ui-sref & ui-sref-active. I can create a directive or other mechanism to make the current menu item highlighted. I thought this should  be part of the menu, hence I am putting a feature request.

**Comment by marcoocram** [comment]
why is broken again? (broken = first default selected xD)

**Comment by cyberlake** [comment]
Current hacky workaround for first element not to be highlighted is to create dummy `menu-item` and set `ng-hide="true"`. But last element still stays highlighted. I tried to add dummy `menu-item` at the end of the list and it didn't help.

**Comment by nicolac18** [comment]
+1

**Comment by oshrilevi** [comment]
+1

**Comment by RolandHeath** [comment]
There is two separate issues here, one of which is resolved as of 0.11.0.

1. The first item is focused when the menu is opened. 
2. (resolved) In 0.10.1 and earlier, multiple items can be highlighted at once just by hovering with the mouse. 

It would be good if the starting focus and mouseleave behaviours were customizable, which I think is what people here are now asking for, although I can foresee some issues as I suspect md-menu-bar uses this element, and needs certain focus behaviours.

**Comment by jlister76** [comment]
Using ui-router's ui-sref is causing the first menu item to remain highlighted while hovering over the second menu item. To temporarily fix this problem I am inserting an empty md-menu-item but would prefer a better solution.

**Comment by Kopleman** [comment]
Have same bug as @jlister76

**Comment by RolandHeath** [comment]
@jlister76 @Kopleman 
If you are hovering another item, can you still change the selection with the up/down arrow keys?
I know that this doesn't solve the bug, but it might indicate whether this is a regression in the presence of ui-sref, or another bug.

**Comment by jlister76** [comment]
@Kopleman @RolandHeath 
Thanks for your input and question as it helped me to realize the true issue and **that the issue is not related to using ui-sref**. The issue is the md-button:focus selector is set to change the background-color of the first button until the down arrow key event. The :focus selector does not change on mouse events. At the same time, the md-button:hover selector is highlighting on hover (as it should). So now, there are two elements being highlighted at the same time. In my opinion, the :focus selector should also correspond to mouse events so that the :focus and :hover can visually be one in the same.

**Comment by cmacdonnacha** [comment]
+1. I have a row of buttons inside**md-menu** and the first button is always highlighted, even when hovering over other buttons.

**Comment by standup75** [comment]
+1

**Comment by Rush** [comment]
Having the same issue. Super confusing.

**Comment by donwitti** [comment]
+1

**Comment by cmacdonnacha** [comment]
I think this needs to be re-opened

**Comment by jayeshanandani** [comment]
I think it's better to open new issue keeping in mind points and template mentioned [here](https://groups.google.com/forum/#!topic/ngMaterial/3Ip9nd68bfk)

**Comment by John-Nagle-IV** [comment]
I worked around this by having a timeout function which would search for the element that had the md-focused class and then remove that class from the element.

**Comment by jasonbaker** [comment]
I ran into this issue when I had the menu content ng-transcluded. A workaround is to manually transclude the menu content.

**Comment by ThomasBurleson** [comment]
We would welcome the a Pull Request  submission from the developer community with fixes for this. At that point we can reopen this issue.

**Comment by marcysutton** [comment]
@ThomasBurleson I seem to remember md-select sets the browser focus along with the mouse cursor, so the focus styles don't persist in the same way. Perhaps that should be considered for menus?

**Comment by ThomasBurleson** [comment]
@all - I am very confused with these conversations. What exactly is the problem ? 

> https://material.angularjs.org/latest/demo/menu highlights the first item and then removes the highlight as you hover over other menu items.

**Comment by marcysutton** [comment]
Ahh, well nevermind then! I don't know what the problem is. Focus still has to be sent into the menu when it opens.

**Comment by zlovin** [comment]
@marcysutton @ThomasBurleson 
When I first opened the issue, I was having trouble with the "normal" use case of md-menu and the first menu item was always retaining highlight, regardless of keyboard input or mouse hover to change the desired menu item. Since then, the issue has been resolved for it's "normal" use case. Therefore, I believe that this issue should remain closed and the issue brought forth by @jasonbaker is separate and unrelated from this issue as it involves a use case outside of "normal" usage.
